refactor: tests for container and cpc

aleyeeey opened the pull request.

aleyeeey committed:
refactor: tests for container and cpc

karsten-klein commented on libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/inspector/JarInspector.java:
```diff
@@ -722,4 +722,14 @@ private String buildPurl(String namespace, String name, String version, String t
         return null;
     }
 
+    private String extractPackageName(String path) {
```
fix artifact.deriveArtifactId

karsten-klein commented on libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/patterns/contributors/ApkPackageContributor.java:
```diff
@@ -131,9 +131,13 @@ public List<ComponentPatternData> contribute(File baseDir, String virtualRootPat
+                            includePatterns.add("usr/share/man/" + packageName + "/**/*");
+                            processCollectedData(components, packageName, version, architecture, includePatterns.toString(), virtualRoot.relativize(relativeAnchorFile).toString(), anchorChecksum, license);
                         }
                     }
```
move process after if-else

karsten-klein commented on libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/patterns/contributors/DotNetComponentPatternContributor.java:
```diff
@@ -120,8 +120,7 @@ private void processProjectFile(List<ComponentPatternData> components, String re
         } catch (Exception e) {
-            // FIXME: adjust to contributor logging and exception handling convention
-            LOG.warn("Failure processing project file", e);
+            LOG.warn("Could not process project file", e);
```
add file path to output; end with .

karsten-klein commented on libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/conan-001/conanfile.py:
```diff
@@ -0,0 +1,37 @@
+from conans import ConanFile, CMake
```
Add header; add link add license; better: obfuscate; reduce to minimum required

karsten-klein commented on libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/go-001/go.mod:
```diff
@@ -0,0 +1,9 @@
+module github.com/skyrocknroll/go-mod-example
```
Add header; add link add license

karsten-klein commented on libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/mix-001/mix.exs:
```diff
@@ -0,0 +1,253 @@
+defmodule Phoenix.MixProject do
```
Add header; add link add license; better: obfuscate; reduce to minimum required

karsten-klein commented on libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/cocoapod-001/HornsAppCore.podspec:
```diff
@@ -0,0 +1,23 @@
+Pod::Spec.new do |spec|
```
Add header; add link add license; better: obfuscate; reduce to minimum required

aleyeeey committed:
Merge branch 'refs/heads/master' into refactor/testsAndContainer

# Conflicts:
#	tests/ae-inventory-subjects-itest/src/test/java/org/metaeffekt/core/itest/analysis/installer/MsiPackageTest.java

aleyeeey committed:
refactor: added requested changes

also refactored tests to download the required files for the tests and removed files in resources

aleyeeey committed:
tests: fixed size of HexTest

karsten-klein commented on libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/patterns/contributors/HexComponentPatternContributor.java:
```diff
@@ -113,9 +113,10 @@ public List<ComponentPatternData> contribute(File baseDir, String virtualRootPat
                         cpd.set(ComponentPatternData.Attribute.VERSION_ANCHOR, relativeAnchorPath);
                         cpd.set(ComponentPatternData.Attribute.VERSION_ANCHOR_CHECKSUM, anchorChecksum);
-                        cpd.set(ComponentPatternData.Attribute.INCLUDE_PATTERN, "**/*");
+                        cpd.set(ComponentPatternData.Attribute.INCLUDE_PATTERN, dependency + "/**/*");
```
Check

aleyeeey committed:
refactor: suggested changes

karsten-klein reviewed (commented):
How does this relate to the other PR; let's review / merge as a whole. Changes in production code must be reviewed with due care.

aleyeeey commented:
@karsten-klein this can be closed, since this is outdated.
